test/midi-util-test.js

```diff
@@ -0,0 +1,104 @@
+var test = require('tape');
+var Util = require('../lib/jsmidgen.js').Util;
+
+test('Midi letter -> pitches', function(t) {
+	var pitch = Util.midi_letter_pitches;
+
+	t.equal(pitch.a, 21, 'Pitch for a should be 21');
+	t.equal(pitch.b, 23, 'Pitch for b should be 23');
+	t.equal(pitch.c, 12, 'Pitch for c should be 12');
+	t.equal(pitch.d, 14, 'Pitch for d should be 14');
+	t.equal(pitch.e, 16, 'Pitch for e should be 16');
+	t.equal(pitch.f, 17, 'Pitch for f should be 17');
+	t.equal(pitch.g, 19, 'Pitch for g should be 19');
+
+	t.end();
+});
+
+// Check pitchFromNote against a wide range to ensure no additional functionality breaks it
+test('Midi pitchFromNote', function(t) {
+	var pitchFromNote = Util.midiPitchFromNote;
+
+	t.equal(pitchFromNote('a1'), 33, 'Pitch of a1 is 33');
+	t.equal(pitchFromNote('b2'), 47, 'Pitch of b2 is 47');
+	t.equal(pitchFromNote('c3'), 48, 'Pitch of c3 is 48');
+	t.equal(pitchFromNote('c#3'), 49, 'Pitch of c#3 is 49');
+	t.equal(pitchFromNote('d4'), 62, 'Pitch of d4 is 62');
+	t.equal(pitchFromNote('e5'), 76, 'Pitch of e5 is 76');
+	t.equal(pitchFromNote('f6'), 89, 'Pitch of f6 is 89');
+	t.equal(pitchFromNote('f#6'), 90, 'Pitch of f#6 is 90');
+	t.equal(pitchFromNote('g7'), 103, 'Pitch of g7 is 103');
+	t.equal(pitchFromNote('g#7'), 104, 'Pitch of g#7 is 104');
+
+	// Check pitch with flattened notes
+	t.equal(pitchFromNote('bb1'), 34, 'Pitch of a#1 is 34');
+	t.equal(pitchFromNote('eb4'), 63, 'Pitch of d#4 is 63');
```

Comment:
That is a very interesting point you brought up. The question I have is, "Is it right for jsmidgen to return some value if you input unconventional notes?" Currently it does. Instinctively I was of the opinion it would throw an error.

(I think we briefly touch upon something similar a while ago)